PMSCore/Admin.cs

```diff
@@ -0,0 +1,35 @@
+﻿namespace PMSCore;
+
+/// <summary>
+/// Represents an Admin user with the highest level of access and permissions.
+/// </summary>
+public class Admin : User
+{
+    /// <summary>
+    /// Initializes an Admin with all permissions.
+    /// </summary>
+    public Admin(string userId)
+        : base(userId, new HashSet<Permission>
+        {
+            Permission.CanCreateProject,
+            Permission.CanChangeProjectProperty,
+            Permission.CanDeleteProject,
+            Permission.CanCreateTask,
+            Permission.CanChangeTaskProperty,
+            Permission.CanDeleteTask,
+            Permission.CanGenerateEntityReport,
+            Permission.CanGenerateUserReport,
+            Permission.CanChangeUser,
+            Permission.CanDeleteUser
+        })
+    {
+    }
+
+    /// <summary>
+    /// Basic validation to ensure Admin-specific functionality can be extended later.
+    /// </summary>
+    public bool ValidateAdminAccess()
+    {
+        return UserPermissions.Contains(Permission.CanGenerateUserReport);
+    }
```

Comment:
If you create an IUser you'll never be able to access this method there is already an I base implementation that generates if the user in question has permissions to do so
https://github.com/DistroRickDev/PMS-Core/blob/37be584ce667f3e979435d344a47816fdac9188a/PMSCore/User.cs#L131